assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -204,7 +204,7 @@ export default function UserDimensionsPieChart( { dimensionName, entityURL, sour
 	}
 
```

Comment:
I noticed a problem while testing, related to this component (see #2425): When I run a development build, the piechart does not show up, in neither the new nor the old widget. I assume that is happening because the chart uses a specific ID but is now used in multiple places?

@eugene-manuilov Do you have an idea how we could fix this? I wonder whether we could somehow increase a static count to prevent duplicate IDs.

Alternatively, if that doesn't work, we could also go with a more pragmatic approach specific to the situation, e.g. `DashboardAllTrafficWidget` could have an optional `isLegacy` prop and based on that the ID used for the inner component's chart would be different - not a great approach, but this situation is temporary anyway, so might be okay. If we can solve it in `UserDimensionsPieChart` internally though, that'd be perfect.

Reply:
> Do you have an idea how we could fix this? I wonder whether we could somehow increase a static count to prevent duplicate IDs.

@felixarntz I think we can leverage the [`useInstanceId`](https://github.com/WordPress/gutenberg/tree/master/packages/compose#useInstanceId) hook to get a unique id for the `UserDimensionsPieChart` component and then add it as a suffix to the `chartID` prop. What do you think?

Reply:
@eugene-manuilov SGTM! cc @benbowler

Reply:
@felixarntz this has been [implemented here](https://github.com/google/site-kit-wp/pull/2617/commits/3add375d269f207bea6b3da62d9933e5a23b1923).